docker-compose.dev.yml

```diff
@@ -1,27 +1,82 @@
 services:
+  nginx:
+    image: nginx:latest
+    ports:
+      - 8080:8080
+    configs:
+      - source: nginx_config
+        target: /etc/nginx/nginx.conf
+    networks:
+      - public
+    depends_on:
+      server:
+        condition: service_healthy
+        restart: true
+
   server:
     build:
       context: ./server
       dockerfile: Dockerfile.dev
-    ports:
-      - '5001:5001'
+    # ports:
+    #   - 5001:5001
     volumes:
       - ./server:/app
       - /app/node_modules
     environment:
       - NODE_ENV=development
     env_file:
       - .env
+    networks:
+      - private
+      - public
+    depends_on:
+      db:
+        condition: service_healthy
+        restart: true
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://server:5001"]
+      interval: 5s
+      retries: 5
+      start_period: 5s
+      timeout: 5s
 
   db:
     image: postgres:latest
-    ports:
-      - '5432:5432'
-    env_file:
-      - .env
+    # ports:
+    #   - "5432:5432"
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_DB: acm_website
+      POSTGRES_PASSWORD_FILE: /run/secrets/pg_password
```

Comment:
postgres related env's are already handled by the `env_file` above

Reply:
secrets are more secure. and i was thinking of having env file only for server to avoid confusion. what do you think?

Reply:
Because this is a docker-compose file that is only used for development on the developers' machine, I think a more convenient and light-weighted setup may be more desirable